plugins/modules/nautobot_server.py

```diff
@@ -0,0 +1,344 @@
+    required: false
+    aliases: [python_path]
+  virtualenv:
+    description:
+      - An optional path to a I(virtualenv) installation to use while running the nautobot-server application.
+    type: path
+    required: false
+    aliases: [virtual_env]
+  db_username:
+    description:
+      - Database username used in Nautobot.
+    type: str
+    required: false
+  db_password:
+    description:
+      - Database password used in Nautobot.
+    type: str
+    required: false
+notes:
+  - This module is inspired from Django_manage Ansible module (U(https://github.com/ansible-collections/community.general/blob/main/plugins/modules/web_infrastructure/django_manage.py)).
+  - To be able to use the C(collectstatic) command, you must have enabled staticfiles in your nautbot_config.py.
+  - Your C(nautobot-server) application must be executable (rwxr-xr-x), and must have a valid shebang, i.e. C(#!/usr/bin/env python), for invoking the appropriate Python interpreter.
+"""
+
+EXAMPLES = r"""
+  - name: Createsuperuser
+    networktocode.nautobot.nautobot_server:
+      command: "createsuperuser"
+      args:
+        email: "admin93@example.com"
+        username: "superadmin7"
+      db_password: "{{ db_password }}"
+  - name: Collectstatic
+    networktocode.nautobot.nautobot_server:
+      command: "collectstatic"
+      db_password: "{{ db_password }}"
+  - name: Post Upgrade
+    networktocode.nautobot.nautobot_server:
+      command: "post_upgrade"
+  - name: Make Migrations for Plugin
+    networktocode.nautobot.nautobot_server:
+      command: "makemigrations"
+      positional_args: ["my_plugin_name"]
+      db_password: "{{ db_password }}"
+  - name: Migrate Plugin
+    networktocode.nautobot.nautobot_server:
+      command: "migrate"
+      args:
+        verbosity: 3
+      flags: ["merge"]
+      positional_args: ["my_plugin_name"]
+      db_username: "{{ db_username }}"
+      db_password: "{{ db_password }}"
+"""
+
+RETURN = r"""
+changed:
+  description: Boolean that is true if the command changed the state.
+  returned: always
+  type: bool
+  sample: True
+out:
+  description: Raw output from the command execution.
+  returned: always
+  type: str
+  sample: superadmin user already exists.
+cmd:
+  description: Full command executed in the Server.
+  returned: always
+  type: str
+  sample: nautobot-server createsuperuser --noinput --email=admin33@example.com --username=superadmin
+project_path:
+  description: The path to the root of the Nautobot application where B(nautobot-server) lives.
+  returned: always
+  type: str
+  sample: /opt/nautobot
+virtualenv:
+  description: An optional path to a I(virtualenv) installation to use while running the nautobot-server application.
+  returned: when defined
+  type: str
+  sample: /opt/nautobot/.venv
+settings:
+  description: The Python path to the application's settings module, such as 'myapp.settings'.
+  returned: when defined
+  type: str
+  sample: myapp.settings
```

Comment:
```suggestion
```

I believe these were removed

Reply:
I kept `virtualenv` as a way to customise the ENV `PATH` is the user has not the `nautobot-server` folder in the `PATH`, but it's optional, so if not provider the `nautobot` user `PATH` should be used.

The same for `settings`, even is not used by any of the current commands, I thought it could be good idea to add it to be used later (together with `pythonpath` that is also not used so far). If you believe that is better to not add `settings` and `pythonpath` until they are actually used, I will remove both.

Reply:
This is in what is returned, not in the available arg spec

Reply:
🤦‍♂️ you are right :)